app/models/piece.rb

```diff
@@ -100,4 +107,13 @@ def is_diagonally_obstructed?(x, y)
       return true
     end
   end
+
+  def has_moved?
+    self.previous_changes["x_coord"].present? || self.previous_changes["y_coord"].present?
```

Comment:
Very nice idea 😄 `previous_changes` seems to be deprecated though and will be no longer available at some point.. Other ideas?

Reply:
I'm having trouble thinking of other solutions.

I looked up the deprecation details for `previous_changes` and from what I understand, the only change will be that the new return value for this method (when called after a change to an attribute but before `save` is called) will reflect its current return value (when `previous_changes` is called after the piece object is saved). Since the `previous_changes` method inside `has_moved?` is only going to be called after the changes to a piece's `x_coord` and `y_coord` have already been saved, I'm assuming the behavior of `previous_changes` will still be the same after the method is updated in the future.

Any ideas as to what direction I should take on this?

For reference:
https://www.ombulabs.com/blog/rails/upgrades/active-record-5-1-api-changes.html
https://github.com/rails/rails/commit/16ae3db5a5c6a08383b974ae6c96faac5b4a3c81

Reply:
@bhk1019 seems like a compelling argument 😄 as an alternative, we could add a flag on pieces, sth like `has_moved` which is false by default but will be set to true after a move.. But I have nothing against the `previous_changes` at this point..

Reply:
What about checking if y_coord == 6 for black or y_coord == 1 for white for has_moved?